variables.tf

```diff
@@ -29,33 +29,31 @@ variable "policy" {
 
 variable "policy_attachment_name" {
   type        = string
-  description = "(Required) - The name of the attachment. This cannot be an empty string."
+  description = "(Optional) - The name of the attachment. Required when users, roles or groups are provided."
+  default     = null
```

Comment:
You may need to revert this back to the initial description but maintain the default value. See the comments that follow for the reason